src/components/NewDatePicker/CalendarPicker/index.js

```diff
@@ -46,16 +47,15 @@ class CalendarPicker extends React.PureComponent {
         if (props.minDate >= props.maxDate) {
             throw new Error('Minimum date cannot be greater than the maximum date.');
         }
-
-        let currentDateView = moment(props.value, CONST.DATE.MOMENT_FORMAT_STRING).toDate();
+        let currentDateView = new Date(props.value);
```

Comment:
There was [another instance](https://github.com/Expensify/App/issues/32535) of date being fetched incorrectly on this line.
When `props.value` type is 'YYYY-MM-DD' and user timezone is behind UTC, the previous date is returned.
`parseISO` should have been used like others.